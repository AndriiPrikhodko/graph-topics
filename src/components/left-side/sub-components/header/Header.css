.header {
    margin-bottom: 5px;
    align-items: center;
}

.title {
    text-align: center;
    font-family: Arial, Helvetica, sans-serif;
    text-decoration: underline #a12110;
    text-decoration-thickness: 3px;
}

.title span {
    color: ghostwhite;
}

.header p {
    text-align: center;
    color: ghostwhite;
}
