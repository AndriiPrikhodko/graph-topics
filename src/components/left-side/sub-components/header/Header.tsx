import './Header.css'
import logo from '../../../../assets/graph.svg'
import IconFactory from '../../../shared/actions/actionIcon.factory'
import Button from '../../../shared/button'
import useLogout from '../../../../helpers/hooks/logout'

const Header = () => {
    const logoutHandler = useLogout()

    return (
        <div className="header">
            <div className='logout-button'>
            <Button label='Logout' onClick={logoutHandler} testid='logout'/>
                <IconFactory iconName='logout' label='logout' onClick={logoutHandler}/>
            </div>
            <div className="logo-title">
                <h3 className="title unselectable">
                    <span>Mind</span> Graph
                </h3>
                <img src={logo} alt="Company Logo" />
            </div>
            <p>Follow the <span>origins</span> of your <span>ideas</span>...</p>
        </div>
    )
}

export default Header