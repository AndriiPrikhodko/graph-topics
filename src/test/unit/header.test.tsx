import {  test } from 'vitest'
import { render } from '@testing-library/react';
// import Graph from '../../components/right-side/graph-engine/graph';
const Graph = import('../../components/right-side/graph-engine/graph')
import Header from '../../components/left-side/sub-components/header/Header';

test('header', () => {
  const { getByRole } = render(<Header />)
  const h3 = getByRole('heading')
  expect(h3).toMatchSnapshot()
})

