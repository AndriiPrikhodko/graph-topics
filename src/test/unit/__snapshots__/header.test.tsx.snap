// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`header 1`] = `
<h3
  class="title unselectable"
>
  <span>
    Graph
  </span>
   Topics
</h3>
`;
